spring-boot/autoconfigure/src/main/java/org/togglz/spring/boot/autoconfigure/TogglzAutoConfiguration.java

```diff
@@ -0,0 +1,251 @@
+            return provider;
+        }
+    }
+
+    @Configuration
+    @ConditionalOnMissingBean(StateRepository.class)
+    protected static class StateRepositoryConfiguration {
+
+        @Autowired
+        private ResourceLoader resourceLoader = new DefaultResourceLoader();
+
+        @Autowired
+        private TogglzProperties properties;
+
+        @Bean
+        public StateRepository stateRepository() throws IOException {
+            StateRepository stateRepository;
+            String featuresFile = properties.getFeaturesFile();
+            if (featuresFile != null) {
+                Resource resource = this.resourceLoader.getResource(featuresFile);
+                Integer minCheckInterval = properties.getFeaturesFileMinCheckInterval();
+                if (minCheckInterval != null) {
+                    stateRepository = new FileBasedStateRepository(resource.getFile(), minCheckInterval);
+                } else {
+                    stateRepository = new FileBasedStateRepository(resource.getFile());
+                }
+            } else {
+                Map<String, FeatureSpec> features = properties.getFeatures();
+                stateRepository = new InMemoryStateRepository();
+                for (String name : features.keySet()) {
+                    stateRepository.setFeatureState(features.get(name).state(name));
+                }
+            }
+            // If caching is enabled wrap state repository in caching state
+            // repository.
+            if (properties.getCache().isEnabled()) {
+                stateRepository = new CachingStateRepository(stateRepository, properties.getCache().getTimeToLive(),
+                        properties.getCache().getTimeUnit());
+            }
+            return stateRepository;
+        }
+    }
+
+    @Configuration
+    @ConditionalOnMissingClass("org.springframework.security.config.annotation.web.configuration.EnableWebSecurity")
+    @ConditionalOnMissingBean(UserProvider.class)
+    protected static class UserProviderConfiguration {
+        @Bean
+        public UserProvider userProvider() {
+            return new NoOpUserProvider();
+        }
+    }
+
+    @Configuration
+    @ConditionalOnClass({ EnableWebSecurity.class, AuthenticationEntryPoint.class, SpringSecurityUserProvider.class })
+    @ConditionalOnMissingBean(UserProvider.class)
+    protected static class SpringSecurityUserProviderConfiguration {
+
+        @Autowired
+        private TogglzProperties properties;
+
+        @Bean
+        public UserProvider userProvider() {
+            return new SpringSecurityUserProvider(properties.getConsole().getFeatureAdminAuthority());
+        }
+    }
+
+    @Configuration
+    @Conditional(TogglzConsoleBaseConfiguration.OnConsoleAndNotUseManagementPort.class)
+    protected static class TogglzConsoleConfiguration extends TogglzConsoleBaseConfiguration {
+
+        public TogglzConsoleConfiguration(@Autowired TogglzProperties properties) {
```

Comment:
No need for `@Autowired` when using constructor injection, it's done implicitly.

Reply:
That's true, I'll fix it.